pkg/cmd/checkendpoints/controller/pod_network_connectivity_check_controller.go

```diff
@@ -97,6 +107,27 @@ func (c *controller) Sync(ctx context.Context, syncContext factory.SyncContext)
 	return nil
 }
 
+// getClientCerts returns the client cert specified in the secret specified in the PodNetworkConnectivityCheck
+// or nil if not specified or there is an error retrieving the certs.
+func (c *controller) getClientCerts(check *operatorcontrolplanev1alpha1.PodNetworkConnectivityCheck) CertificatesGetter {
+	return func() []tls.Certificate {
+		if len(check.Spec.TLSClientCert.Name) > 0 {
+			secret, err := c.secretLister.Secrets(c.podNamespace).Get(check.Spec.TLSClientCert.Name)
+			if err != nil {
+				klog.V(2).Infof("secret/%s: %v", check.Spec.TLSClientCert.Name, err)
+				return nil
+			}
+			cert, err := tls.X509KeyPair(secret.Data["tls.crt"], secret.Data["tls.key"])
+			if err != nil {
+				klog.V(2).Infof("error loading tls client key pair: %v", err)
```

Comment:
event worthy?  this shouldn't happen. Of course, writing events would be a big event write load.  

//TODO write a rate limited event every minute.